github_api_toolkit/__init__.py

```diff
@@ -155,3 +153,162 @@ def post(self, url: str, params: dict = {}, add_prefix: bool = True) -> requests
+            list | tuple: A list of verified domain emails for the user in the organization or a tuple containing an error message and status code.
+        """
+        
+        self.query = '''
+            query ($username: String!, $org: String!) {
+                user (login: $username) {
+                    login
+                    organizationVerifiedDomainEmails(login: $org)
+                }
+            }
+        '''
+
+        self.params = {
+            'username': username,
+            'org': org
+        }
+
+        response = self.make_ql_request(self.query, self.params)
+
+        if response.status_code == 200:
+            return response.json()["data"]["user"]["organizationVerifiedDomainEmails"]
+        else:
+            response_json = response.json()
+            return response_json["message"], response_json["status"]
+        
+    def get_codeowner_teams(self, org: str, repo: str) -> list | tuple:
+        """Gets the CODEOWNERS file from a GitHub repository and returns the teams listed in the file.
+
+        Args:
+            org (str): the GitHub organization name.
+            repo (str): the GitHub repository name.
+
+        Returns:
+            list | tuple: A list of teams listed in the CODEOWNERS file or a tuple containing an error message and status code.
+        """
+
+        self.query = '''
+            query ($org: String!, $repo: String!) {
+                repository(owner: $org, name: $repo) {
+                    file: object(expression: "main:CODEOWNERS") {
+                        ... on Blob {
+                            text
+                        }
+                    }
+                }
+            }
+        '''
+
+        self.params = {
+            'org': org,
+            'repo': repo
+        }
+
+        response = self.make_ql_request(self.query, self.params)
+
+        if response.status_code == 200:
+            codeowners_contents = response.json()["data"]["repository"]["file"]["text"]
+
+            codeowners = codeowners_contents.split("\n")
+
+            # Remove any empty strings from the list of codeowners
+
+            indexes_removed = 0
+
+            for i in range(0, len(codeowners)):
+                if codeowners[i - indexes_removed] == "":
+                    codeowners.pop(i - indexes_removed)
+                    indexes_removed += 1
+
+            for i in range(len(codeowners)):
+                # If the codeowner contains a "/", it is a team and needs to be split from the organisation name.
+                # If there is no "/", it is a user and needs to have the @ symbol removed.
+                if "/" in codeowners[i]:
+                    codeowners[i] = codeowners[i].split("/")[-1]
+                else:
+                    codeowners[i] = codeowners[i].replace("@", "")
+
+            return codeowners
+        else:
+            response_json = response.json()
+            return response_json["message"], response_json["status"]
```

Comment:
Is there a danger that the response may not include message? or status?
It may be safer with the following:

return response_json.get("message", "No message"), response_json.get("status", "Unknown status")